### PR TITLE
Added last CannotUseItemsInBattle tests

### PR DESCRIPTION
**Pull request by LOuroboros**
## Description
This PR adds the last tests and `TO_DO` tests needed to review #3524.
Tests that have been added as `TO_DO` are tests that either aren't possible to write right now, or that I don't know if they *can* be written currently.

## **Discord contact info**
lunos4026

### CODE REVIEW

File: test/battle/item_effect/escape.c [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#include "global.h"
+#include "test/battle.h"
+
+WILD_BATTLE_TEST("Poke Toy lets the player escape from a wild battle")
+{
+    GIVEN {
+        ASSUME(gItems[ITEM_POKE_TOY].battleUsage == EFFECT_ITEM_ESCAPE);
```
Comment (AlexOn1ine) [inline test/battle/item_effect/escape.c:7]: Can you replace all of the tests assumes with one at the top?

```
ASSUMPTIONS
{
    ASSUME(gItems[ITEM_POKE_TOY].battleUsage == EFFECT_ITEM_ESCAPE);
}
```

Comment (LOuroboros) [reply test/battle/item_effect/escape.c]: Done.

File: test/battle/item_effect/escape.c [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    }
+}
+
+WILD_BATTLE_TEST("Poke Toy lets the player escape from a wild battle even if a move forbid them to")
+{
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_MEAN_LOOK); }
```
Comment (AlexOn1ine) [inline test/battle/item_effect/escape.c:27]: Sorry, missed that one. Please add an assume for the mean look effect inside this test.

Comment (LOuroboros) [reply test/battle/item_effect/escape.c:28]: Done.

**Review by AlexOn1ine — APPROVED** [review]

**Pull request merged**